### PR TITLE
enable asan in remaining rocm packages

### PR DESCRIPTION
**Pull request by afzpatel**

### CODE REVIEW

File: var/spack/repos/builtin/packages/migraphx/package.py [hunk 1]
```diff
@@ -134,6 +145,12 @@ def cmake_args(self):
             args.append(
                 self.define("GPU_TARGETS", "gfx906;gfx908;gfx90a;gfx1030;gfx1100;gfx1101;gfx1102")
             )
+        if self.spec.satisfies("@6.1:") and self.spec.satisfies("+asan"):
+            args.append(
+                self.define(
+                    "CMAKE_CXX_FLAGS", "-fsanitize=address -shared-libasan -I{0}".format(abspath)
```
Comment (srekolam) [inline var/spack/repos/builtin/packages/migraphx/package.py:155]: this is added thru  env.set("CXXFLAGS", "-fsanitize=address -shared-libasan") ? and abspath is added from 5.5.0 onwards L140.

**Review by srekolam — APPROVED** [review]

File: var/spack/repos/builtin/packages/migraphx/package.py (hunk 1 above)
Comment (afzpatel) [reply var/spack/repos/builtin/packages/migraphx/package.py:155]: L140 overides the CXX_FLAGS previously set by env.set("CXXFLAGS", "-fsanitize=address -shared-libasan"), this is why the flags need to be set again.

**Review by renjithravindrankannath — COMMENTED** [review]
asan variant can be restricted to rhel-8, sles, ubuntu-20 and ubuntu-22

**Review by renjithravindrankannath — APPROVED** [review]
LGTM

**Comment by afzpatel** [comment]
@spackbot rerun pipeline

**Comment by afzpatel** [comment]
@spackbot rerun pipeline

**Comment by afzpatel** [comment]
@spackbot rerun pipeline

**Comment by afzpatel** [comment]
@spackbot rerun pipeline

**Comment by afzpatel** [comment]
Hi @tldahlgren, this pr is ready to be reviewed

**Review by tldahlgren — APPROVED** [review]

**Pull request merged**